core/src/main/java/org/pentaho/platform/util/oauth/PentahoOAuthUtility.java

```diff
@@ -0,0 +1,166 @@
+
+  private static boolean isProviderOAuth = false;
+
+  private static boolean isProviderDualAuth = false;
+
+  private static PentahoOAuthUtility instance;
+
+  // Private constructor to prevent instantiation from outside
+  private PentahoOAuthUtility() {
+  }
+
+  // Public method to provide access to the single instance
+  public static PentahoOAuthUtility getInstance() {
+    if ( instance == null ) {
+      instance = new PentahoOAuthUtility();
+    }
+    return instance;
+  }
+
+  public static boolean isOAuthEnabled() {
+
+    IConfiguration config = systemConfig.getConfiguration( "security" );
+
+    if ( !isProviderOAuth ) {
+      try {
+        String provider = config.getProperties().getProperty( "provider" );
+        isProviderOAuth = StringUtils.equals( provider, OAUTH );
+        isProviderDualAuth = StringUtils.equals( provider, DUALAUTH );
+      } catch ( IOException e ) {
+        isProviderOAuth = false;
+      }
+    }
+
+    if ( !isProviderOAuth && isProviderDualAuth ) {
+      config = systemConfig.getConfiguration( OAUTH );
+      try {
+        String provider1 = config.getProperties().getProperty( "provider1" );
+        String provider2 = config.getProperties().getProperty( "provider2" );
+        isProviderOAuth = StringUtils.equals( provider1, OAUTH ) || StringUtils.equals( provider2, OAUTH );
+      } catch ( IOException e ) {
+        isProviderOAuth = false;
+      }
+    }
+
+    return isProviderOAuth;
+  }
+
+  public static boolean isOAuthEnabledWithDualAuth() {
+
+    IConfiguration config = systemConfig.getConfiguration( "security" );
+
+    if ( !isProviderDualAuth ) {
+      String provider = null;
+      try {
+        provider = config.getProperties().getProperty( "provider" );
+      } catch ( IOException e ) {
+        isProviderDualAuth = false;
+      }
+      isProviderDualAuth = StringUtils.equals( provider, DUALAUTH );
+    }
+
+    if ( isProviderDualAuth ) {
+      config = systemConfig.getConfiguration( DUALAUTH );
+      String provider1 = null;
+      String provider2 = null;
+      try {
+        provider1 = config.getProperties().getProperty( "provider1" );
+        provider2 = config.getProperties().getProperty( "provider2" );
+      } catch ( IOException e ) {
+        isProviderOAuth = false;
+      }
+      isProviderOAuth = StringUtils.equals( provider1, OAUTH ) || StringUtils.equals( provider2, OAUTH );
+    }
+
+    return isProviderOAuth;
+  }
+
+  public static boolean shouldPerformLiveUpdate() {
+    IConfiguration config = systemConfig.getConfiguration( OAUTH );
+    try {
+      return Boolean.parseBoolean( config.getProperties().getProperty( "shouldPerformLiveUpdate", "false" ) );
+    } catch ( IOException e ) {
+      return false;
+    }
+  }
+
+  public static boolean isUserNamePasswordAuthentication( HttpServletRequest httpServletRequest ) {
+    return "/pentaho/j_spring_security_check".equals( httpServletRequest.getRequestURI() ) ||
```

Comment:
The URL may not start with "pentaho". The customer might have configured a different app name.

Reply:
This might not even be "j_spring_security_check" either. This should be pulled form the spring configuration.